res/controllers/Traktor-Kontrol-Z1-scripts.js

```diff
@@ -0,0 +1,204 @@
+            this.linkKnob('default', channel + '_mid', '[EqualizerRack1_[Channel' + channel + ']_Effect1]', 'parameter2')
+            this.linkKnob('default', channel + '_low', '[EqualizerRack1_[Channel' + channel + ']_Effect1]', 'parameter1')
+            this.linkKnob('default', channel + '_fx', '[QuickEffectRack1_[Channel' + channel + ']]', 'super1')
+
+            controller.setCallback('control', 'hid', channel + '_gain', this.knob)
+            controller.setCallback('control', 'hid', channel + '_hi', this.knob)
+            controller.setCallback('control', 'hid', channel + '_mid', this.knob)
+            controller.setCallback('control', 'hid', channel + '_low', this.knob)
+            controller.setCallback('control', 'hid', channel + '_fx', this.knob)
+
+            this.linkKnob('default', channel + '_vol', '[Channel' + channel + ']', 'volume')
+            controller.setCallback('control', 'hid', channel + '_vol', this.knob)
+        }
+
+        this.linkKnob('default', 'cue_mix', '[Master]', 'headMix')
+        controller.setCallback('control', 'hid', 'cue_mix', this.knob)
+        this.linkKnob('default', 'crossfader', '[Master]', 'crossfader')
+        controller.setCallback('control', 'hid', 'crossfader', this.knob)
+    }
+
+    // endregion
+
+    // region LIGHTS
+
+    this.registerOutputPackets = function () {
+        var packet = new HIDPacket('lights', [0x80], 22)
+
+        for (var c = 1; c < 3; c++) {
+            for (var i = 1; i < 8; i++) {
+                packet.addControl('hid', 'ch' + c + '_segment' + i, i + (c - 1) * 7, 'B')
+            }
+            packet.addControl('hid', c + '_headphone', 14 + c, 'B')
+            packet.addControl('hid', c + '_button_fx_red', 14 + c * 3, 'B')
+            packet.addControl('hid', c + '_button_fx_blue', 15 + c * 3, 'B')
+        }
+        packet.addControl('hid', 'mode', 19, 'B')
+
+        this.controller.registerOutputPacket(packet)
+    }
+
+    this.sendLightsUpdate = function () {
+        this.controller.getLightsPacket().send()
+    }
+
+    this.brightness = 0x7f
+    this.brightnessRange = 1.0 / 7
+    this.refreshVolumeLights = function (value, group, key) {
+        var packet = this.controller.getLightsPacket()
+        const channel = group.substr(8, 1)
+        for (var i = 0; i < 7; i++) {
+            var br = Math.max(Math.min((value - i * this.brightnessRange) * 7, 1), 0) * this.brightness
+            packet.getField('hid', 'ch' + channel + '_segment' + (i + 1)).value = br
+        }
+        this.sendLightsUpdate()
+    }
+
+    // endregion
+
+}
+
+var KontrolZ1 = new KontrolZ1Controller()
+
+KontrolZ1.init = function (id) {
+    KontrolZ1.id = id
+
+    KontrolZ1.registerInputPackets()
+    KontrolZ1.registerOutputPackets()
+    KontrolZ1.registerCallbacks()
+
+    for (var c = 1; c < 3; c++) {
+        engine.connectControl('[Channel' + c + ']', 'VuMeter', KontrolZ1.refreshVolumeLights)
```

Comment:
does it have the same syntax? Can't test right now

Reply:
It has the same syntax except dropping support for the hacky old syntax passing a string of JS code instead of a function, which you do not use so it should be fine.